continuous-integration/readme.md

```diff
@@ -1,3 +1,233 @@
+
+- [ ] **Everyone commits to the git repository each day**
+  - End of day checked-in code should contain unit tests at the minimum.
+  - Run the build locally before checking in to avoid CI pipeline failure saturation. You should verify what caused the error, and try to solve it as soon as possible instead of committing your code. We encourage developers to follow a [lean SDLC principles](https://leankit.com/learn/lean/principles-of-lean-development/).
+  - Isolate work into small chunks which ties directly to business value and refactor incrementally.
+
+## Isolated Environments
+
+One of the key goals of build validation is to isolate and identify failures in staging environment(s) and minimize any disruption to live production traffic. Our E2E automated tests should run in an environment which mimics our production environment(as much as possible). This includes consistent software versions, OS, test data volume simulations, network traffic parity with production, etc.
+
+- [ ] **Test in a clone of production**
+  - The production environment should be duplicated into a staging environment(QA and/or Pre-Prod) at a minimum.
+- [ ] **Pull request update(s) trigger staged releases**
+  - New commits related to a pull request should trigger a build / release into an integration environment. The production environment should be fully isolated from this process.
+- [ ] **Promote infrastructure changes across fixed environments**
+  - Infrastructure as code changes should be tested in a integration environment and promoted to all staging environment(s) then migrated to production with zero downtime for system users.
+- [ ] **Environment releases are triggered on merges into master**
+  - Code changes merged into the master branch should auto trigger a new release candidate for all cloud environments.
+  - One additional capability worth considering are automated rollbacks. Being able to revert commits into master and quickly go back to the last known good state(*previous feature release*) minimizes outages and downtime when bugs are introduced into environment(s).
+- [ ] **Testing in production**
+  - There are various [approaches](https://medium.com/@copyconstruct/testing-in-production-the-safe-way-18ca102d0ef1) with safely carrying out automated tests for production deployments. Some of these may include:
+    - Feature flagging
+    - A/B testing
+    - Traffic shifting
+
+## Developer Access to Latest Release Artifacts
+
+Our devops workflow should enable developers to get, install and run the latest system executable. Release executable(s) should be auto generated as part of our CI/CD pipeline(s).
+
+- [ ] **Developers can access latest executable**
+  - Latest system executable is available for all developers on the team. There should be a well-known place where developers can reference the release artifact.
+
+- [ ] **Release artifact is published for each git commit per open pull request or merges into master**
```

Comment:
Should this be **Release artifact is published for each pull request or merges into master**.  Also should you mention following semver versioning?